lib/dialects/sqlite/connection-manager.js

```diff
@@ -59,8 +59,7 @@ class ConnectionManager extends AbstractConnectionManager {
         options.readWriteMode || this.lib.OPEN_READWRITE | this.lib.OPEN_CREATE, // default mode
         err => {
           if (err) {
-            if (err.code === 'SQLITE_CANTOPEN') return reject(new sequelizeErrors.ConnectionError(err));
-            return reject(new sequelizeErrors.ConnectionError(err));
+            reject(new sequelizeErrors.ConnectionError(err));
```

Comment:
Missing return? This would still trigger the resolver (basically a no-op).

Reply:
Right, but when it triggers the resolver, lines 62 and 63 return undefined, and with the return, it returns "undefined" instead of the promise, which results in a `cannot read property 'then' of undefined'. 

I'll update the test cases (even with master I'm still getting errors about the DB being locked on some tests) but I don't know why there are two conditions in the case which return exactly the same case - if the error code is can't open return x, if it isn't, return x.